test/parallel/test-http-hostname-typechecking.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const vals = [{}, [], NaN, Infinity, -Infinity, true, false, 1, new Date()];
+
+vals.forEach((v) => {
+  assert.throws(
+    () => http.request({hostname: v}),
+    /^TypeError: "hostname" must either be a string, undefined or null$/);
+});
+
+['', undefined, null].forEach((v) => {
+  assert.doesNotThrow(() => {
+    http.request({hostname: v}).on('error', common.noop);
```

Comment:
The main part that this is testing is that the call to `http.request()` does not throw synchronously. The fact that it will emit the `error` event is non-sequential to this particular test. This is set simply to avoid the unhandled `error` exception.